packages/reactive-element/src/decorators/query.ts

```diff
@@ -64,6 +84,19 @@ export function query(selector: string, cache?: boolean): QueryDecorator {
     descriptor?: PropertyDescriptor
   ) => {
     const doQuery = (el: Interface<ReactiveElement>): V => {
+      if (DEV_MODE && !el.hasUpdated) {
```

Comment:
To resolve the test failure, and to be more specific for the caching `null` case, I propose adding an `&& cache` here.

Reply:
Interesting. The warning itself is generic that the getter was called before first update, and not specific to the `cache` option.

I wonder if the failing task test should be updated instead to exclude this warning instead of opting out this warning. (also interesting to see that the task test does highlight one way the query decorated property might get accessed before first update).

Reply:
I could also be convinced to just only warn when `cache` is true too. It's more impactful to raise attention if `null` is getting cached where as the warning for none-caching scenarios might just be excess noise.

Reply:
Given that a test has shown a valid reason to access a `@query` decorated field before the first update (in a Task dependency array). I wouldn't want to silence the warning in only that test case, because we may end up having users that get the warning via a valid usage of `Task` depending on a `@query` field.

I think the warning makes sense if there is something bad going on, but the fix to `cache` behavior resolves the worst possible bugs. I propose removing the warnings from this PR.

Reply:
I agree, it seems fine to not warn when there's a legit case to encounter it. @rictic originally suggested in the issue and PR to add a dev mode warning, what do you think about removing it?

Reply:
There are some combinations I've considered for the warning. We could issue it if `cache` is true, but also when the query result is actually `null`. Would that be acceptable?

Reply:
Caching null sounds like an unintended behavior so warning in that case sounds reasonable to me.

Reply:
Yes please. Making the warning specific to: `cache: true` and result is `null`, is a perfect time for a warning. It's much more specific to the issue you are fixing.
Thanks!